integrations/aws/utils.py

```diff
@@ -0,0 +1,189 @@
+    """
+    Checks if the resource kind is a global resource
+    """
+    global_services = ['cloudfront', 'route53', 'waf', 'waf-regional', 'iam', 'organizations']
+    service = kind.split('::')[1].lower()
+    return service in global_services
+
+def _fix_unserializable_date_properties(obj: Any) -> Any:
+    """
+    Handles unserializable date properties in the JSON by turning them into a string
+    """
+    return json.loads(json.dumps(obj, default=str))
+
+async def describe_single_resource(kind: str, identifier: str, account_id: Optional[str] = None, region: Optional[str] = None) -> dict[str, Any]:
+    """
+    Describes a single resource using the CloudControl API.
+    """
+    async for session in _get_sessions(account_id, region):
+        region = session.region_name
+        try:
+            if kind == ResourceKindsWithSpecialHandling.ACM:
+                async with session.client('acm') as acm:
+                    response = await acm.describe_certificate(CertificateArn=identifier)
+                    return response.get('Certificate', {})
+            
+            elif kind == ResourceKindsWithSpecialHandling.LOADBALANCER:
+                async with session.client('elbv2') as elbv2:
+                    response = await elbv2.describe_load_balancers(LoadBalancerArns=[identifier])
+                    return response.get('LoadBalancers', [])[0]
+            
+            elif kind == ResourceKindsWithSpecialHandling.CLOUDFORMATION:
+                async with session.client('cloudformation') as cloudformation:
+                    response = await cloudformation.describe_stacks(StackName=identifier)
+                    return response.get('Stacks', [])[0]
+            
+            elif kind == ResourceKindsWithSpecialHandling.EC2:
+                async with session.client('ec2') as ec2_client:
+                    described_instance = await ec2_client.describe_instances(InstanceIds=[identifier])
+                    instance_definition = described_instance["Reservations"][0]["Instances"][0]
+                    return instance_definition
+                
+            else:
+                async with session.client('cloudcontrol') as cloudcontrol:
+                    response = await cloudcontrol.get_resource(TypeName=kind, Identifier=identifier)
+                    resource_description = response.get('ResourceDescription', {})
+                    return {
+                        'Identifier': resource_description.get('Identifier', ''),
+                        'Kind': kind,
+                        **json.loads(resource_description.get('Properties', {}))
+                    }
+        except Exception as e:
+            if e.response['Error']['Code'] == 'ResourceNotFoundException':
+                logger.info(f"Resource not found: {kind} {identifier}")
+                return {}
+            logger.error(f"Failed to describe CloudControl Instance in region: {region}; error {e}")
```

Comment:
why not raising the error and handling it outside?

Reply:
Because an unfound resource is not an error, for example resources that were deleted.
also it might complicate the code a bit since there will be multiple nested `try catches` in the main functions